sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -178,19 +211,25 @@ public Response create(Object document) {
                 return res;
             }
         } catch (CouchException e) {
-            if (COUCH_ERROR_CONFLICT.equals(e.getError())) {
+            if (e.getStatusCode() == 409) {
                 throw new DocumentConflictException(e.toString());
+            } else {
+                throw e;
             }
-            throw e;
         } finally {
             closeQuietly(is);
         }
     }
 
+    // TODO TODO TODO
+    // slim down the number of methods here, we don't need all of these variants
+    // make revid null if we don't want to specify it; bool flag for compressed/uncompressed atts
+
```

Comment:
Some other time; it's nice to be able to eyeball the changes here don't change things.